sdk/eventhub/event-hubs/src/eventHubSender.ts

```diff
@@ -550,6 +553,9 @@ export class EventHubSender extends LinkEntity {
           this.isOpen(),
           this.isConnecting
         );
+        return this._sender;
+      } else {
+        throw new MessagingError("The sender link could not be opened.");
```

Comment:
When do expect this to be hit?

Reply:
Honestly, never. In code coverage I don't see this being hit. It's more of a CYA in case it does happen, and because TypeScript will complain if I don't throw an error or return the sender in this case.

Reply:
Say `_sender` is undefined, init() gets called which sets isConnecting to true.
Is it possible that before `createAwaitableSender` completes, another init() request gets called?
In that case, previously we would have simply returned and now we would throw this error.

Reply:
It is not possible for another init() request to get called because we wrap all init requests in acquireLock. So init won't be called until the previous init has completed. (And this is the crux of why I say this shouldn't be called outside of `_createLinkIfNotOpen()`

Reply:
Then, what is the point of tracking `isConnecting`?

Reply:
Cool, if that works out and you can replace `if (!this.isOpen() && !this.isConnecting)` with `if (!this.isOpen() || !this._sender)`, then you can remove the rest of the changes to this if/else block

Reply:
Done! That cleaned up _init nicely.

Reply:
Awesome, thanks!